evennia/objects/objects.py

```diff
@@ -1656,20 +1659,25 @@ def return_appearance(self, looker, **kwargs):
             **kwargs (dict): Arbitrary, optional arguments for users
                 overriding the call (unused by default).
         """
+        def filter_visible(obj_list):
+            # Helper method to determine if objects are visible to the looker.
+            return [obj for obj in obj_list if obj != looker and obj.access(looker, "view")]
+
         if not looker:
             return ""
+
         # get and identify all objects
-        visible = (con for con in self.contents if con != looker and con.access(looker, "view"))
-        exits, users, things = [], [], defaultdict(list)
-        for con in visible:
-            key = con.get_display_name(looker)
-            if con.destination:
-                exits.append(key)
-            elif con.has_account:
-                users.append("|c%s|n" % key)
-            else:
-                # things can be pluralized
-                things[key].append(con)
+        exits_list = filter_visible(self.contents_get(category='exit'))
+        users_list = filter_visible(self.contents_get(category='character'))
+        things_list = filter_visible(self.contents_get(category="object"))
+
+        things = defaultdict(list)
+
+        for thing in things_list:
+            things[thing.key].append(thing)
+        users = [f"|c{user.key}|n" for user in users_list]
+        exits = [ex.key for ex in exits_list]
+
         # get description, build string
```

Comment:
Looks like the base `Object` is missing a `_content_types` tuple. It seems reasonable that you should be able to filter based on 'plain' "object" type as well.